src/domain/order/use-cases/find-product-by-category.ts

```diff
@@ -0,0 +1,35 @@
+import { NotFoundError } from '@cloud-burger/handlers';
+import logger from '@cloud-burger/logger';
+import { Product } from '../entities/product';
+import { ProductRepository } from '../repositories/product';
+import { ProductCategory } from '../entities/value-objects/enums/product-category';
+
+interface Input {
+    category: string;
+}
+
+export class FindProductByCategoryUseCase {
+    constructor(private productRepository: ProductRepository) {}
+
+    async execute({ category }: Input): Promise<Product[]> {
```

Comment:
```suggestion
    async execute({ category }: Input): Promise<Product[]> {
        return await this.productRepository.findByCategory(category);
    }
```
Aqui acredito que possa retornar direto (vai retornar ou o array de produtos ou array vazio)